src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -30,5 +33,27 @@ Task ISocketOutput.WriteAsync(ArraySegment<byte> buffer, bool immediate, Cancell
             _outputStream.Write(buffer.Array, buffer.Offset, buffer.Count);
             return TaskUtilities.CompletedTask;
         }
+
+        public MemoryPoolIterator2 ProducingStart()
+        {
+            _producingBlock = _memory.Lease();
+            return new MemoryPoolIterator2(_producingBlock);
+        }
+
+        public void ProducingComplete(MemoryPoolIterator2 end, int count)
+        {
+            var block = _producingBlock;
+            while (block != end.Block)
+            {
+                _outputStream.Write(block.Data.Array, block.Data.Offset, block.Data.Count);
```

Comment:
`StreamSocketOutput.WriteAsync` also does sync writes. There is a TODO comment to use WriteAsync. We need to figure out why that breaks HTTPS.